### PR TITLE
Adding to the tour book and calendar the average elevation change per unit of distance

### PR DESCRIPTION
**Pull request by FJBDev**

### CODE REVIEW

File: bundles/net.tourbook/src/net/tourbook/data/TourData.java [hunk 1]
```diff
@@ -355,6 +355,13 @@
    @XmlElement
    private int                   tourAltDown;
 
+
+   /**
+    * Average altitude change (m/km)
+    */
+   @XmlElement
+   private int                   tourAvgAltChange;
```
Comment (wolfgang-ch) [inline bundles/net.tourbook/src/net/tourbook/data/TourData.java]: The field `tourAvgAltChange` should have a better name.

In the beginning of MT all fields started with `tour...` however all fields are just for one tour in `TourData` and `Alt` should be named without abbreviation to be easier to read. `Avg` is OK as it is commonly used.

So my suggestion is `avgAltitudeChange`

File: bundles/net.tourbook/src/net/tourbook/ui/views/tourBook/TourBookView.java [hunk 2]
```diff
@@ -962,10 +959,13 @@ private void defineAllColumns(final Composite parent) {
       defineColumn_Altitude_Up();
       defineColumn_Altitude_Down();
       defineColumn_Altitude_Max();
+      defineColumn_Avg_Altitude_Change();
```
Comment (wolfgang-ch) [inline bundles/net.tourbook/src/net/tourbook/ui/views/tourBook/TourBookView.java]: All altitude columns should start with `defineColumn_Altitude`

My suggestion is `defineColumn_Altitude_AvgChange`

File: bundles/net.tourbook/src/net/tourbook/ui/views/tourBook/TVITourBookItem.java [hunk 3]
```diff
@@ -146,6 +149,7 @@
 
    long         colAltitudeUp;
    long         colAltitudeDown;
+   float        colAvgAltitudeChange;
```
Comment (wolfgang-ch) [inline bundles/net.tourbook/src/net/tourbook/ui/views/tourBook/TVITourBookItem.java]: My suggestion for the name is `colAltitude_AvgChange` similar to `colTemperature_...`

File: bundles/net.tourbook/src/net/tourbook/ui/views/tourBook/TVITourBookYearSub.java [hunk 4]
```diff
@@ -270,6 +286,7 @@ protected void fetchChildren() {
                final long dbDrivingTime         = tourItem.colTourDrivingTime = result.getLong(6);
                tourItem.colAltitudeUp           = result.getLong(7);
                tourItem.colAltitudeDown         = result.getLong(8);
+               tourItem.colAvgAltitudeChange    = result.getLong(84);
```
Comment (wolfgang-ch) [inline bundles/net.tourbook/src/net/tourbook/ui/views/tourBook/TVITourBookYearSub.java]: reading field 84 should not be out of the sequence.

`colTemperature_Min/Max` is also read at the end and not where `colTemperature_Avg` is located, in this case the sequence is more important than the grouping

File: bundles/net.tourbook/src/net/tourbook/ui/TreeColumnFactory.java [hunk 5]
```diff
@@ -199,6 +202,26 @@ public TreeColumnDefinition createColumn(final ColumnManager columnManager,
          }
       };
 
+      AVG_ALTITUDE_CHANGE = new TreeColumnFactory() {
```
Comment (wolfgang-ch) [inline bundles/net.tourbook/src/net/tourbook/ui/TreeColumnFactory.java]: My suggestion `ALTITUDE_AVG_CHANGE`

File: bundles/net.tourbook/src/net/tourbook/ui/TreeColumnFactory.java [hunk 6]
```diff
@@ -199,6 +202,26 @@ public TreeColumnDefinition createColumn(final ColumnManager columnManager,
...
+                                                  final PixelConverter pixelConverter) {
+
+            final TreeColumnDefinition colDef = new TreeColumnDefinition(columnManager, "AVG_ALTITUDE_CHANGE", SWT.TRAIL); //$NON-NLS-1$
+            final String unitLabel = UI.SYMBOL_AVERAGE + UI.SPACE + UI.UNIT_LABEL_ALTITUDE + "/" + UI.UNIT_LABEL_DISTANCE; //$NON-NLS-1$
+
+            colDef.setColumnCategory(Messages.ColumnFactory_Category_Altitude);
+            colDef.setColumnLabel(Messages.ColumnFactory_avg_altitude_change_label);
+            colDef.setColumnHeaderText(unitLabel);
+            colDef.setColumnUnit(unitLabel);
+            colDef.setColumnHeaderToolTipText(Messages.ColumnFactory_avg_altitude_change_tooltip);
```
Comment (wolfgang-ch) [inline bundles/net.tourbook/src/net/tourbook/ui/TreeColumnFactory.java]: My suggestion `ColumnFactory_Altitude_AvgChange_Label/Tooltip` similar to 
`ColumnFactory_Temperature_ ... _Label/_Tooltip`

File: bundles/net.tourbook/src/net/tourbook/data/TourData.java (hunk 1 above)
Comment (FJBDev) [reply bundles/net.tourbook/src/net/tourbook/data/TourData.java]: Done (see PR)

File: bundles/net.tourbook/src/net/tourbook/ui/views/tourBook/TourBookView.java (hunk 2 above)
Comment (FJBDev) [reply bundles/net.tourbook/src/net/tourbook/ui/views/tourBook/TourBookView.java]: Done (see PR)

File: bundles/net.tourbook/src/net/tourbook/ui/views/tourBook/TVITourBookYearSub.java (hunk 4 above)
Comment (FJBDev) [reply bundles/net.tourbook/src/net/tourbook/ui/views/tourBook/TVITourBookYearSub.java]: Done (see PR)

File: bundles/net.tourbook/src/net/tourbook/ui/TreeColumnFactory.java (hunk 5 above)
Comment (FJBDev) [reply bundles/net.tourbook/src/net/tourbook/ui/TreeColumnFactory.java]: Done (see PR)

File: bundles/net.tourbook/src/net/tourbook/ui/TreeColumnFactory.java (hunk 6 above)
Comment (FJBDev) [reply bundles/net.tourbook/src/net/tourbook/ui/TreeColumnFactory.java]: Done (see PR)

**Pull request merged**